pymodbus/framer/rtu_framer.py

```diff
@@ -236,3 +304,8 @@ def recvPacket(self, size):
         result = self.client.recv(size)
         self.client.last_frame_end = round(time.time(), 6)
         return result
+
+    def __del__(self):
+        """Cancel the worker task if needed."""
+        if self._worker is not None:
+            self._worker.cancel()
```

Comment:
This is not sufficient, you need to wait until it is actually cancelled. Especially windows leaves tasks hanging for quite a while.